drivers/hmis/app/models/hmis/hud/hmis_service.rb

```diff
@@ -16,9 +16,43 @@ class Hmis::Hud::HmisService < Hmis::Hud::Base
 
   belongs_to :owner, polymorphic: true # Service or CustomService
   belongs_to :custom_service_type
+  has_many :custom_service_categories, through: :custom_service_type
+
   alias_attribute :service_type, :custom_service_type
+  alias_to_underscore [:DateProvided]
 
   SORT_OPTIONS = [:date_provided].freeze
+  HUD_ATTRIBUTES = [:record_type, :type_provided, :other_type_provided, :moving_on_other_type, :sub_type_provided, :referral_outcome, :FAAmount].freeze
+
+  HUD_ATTRIBUTES.each do |hud_field_name|
+    define_method(hud_field_name) { hud_service&.send(hud_field_name) }
+  end
+
+  scope :in_service_category, ->(category_id) do
+    type_ids = Hmis::Hud::CustomServiceType.where(custom_service_category_id: category_id).pluck(:id)
+    where(custom_service_type_id: type_ids)
+  end
+
+  scope :matching_search_term, ->(search_term) do
+    return none unless search_term.present?
+
+    search_term.strip!
+    query = "%#{search_term}%"
+    joins(:custom_service_type, :custom_service_categories).
+      where(cst_t[:name].matches(query).or(csc_t[:name].matches(query)))
+  end
+
+  def id
+    "#{owner_type.sub('Hmis::Hud::', '')}:#{owner_id}"
+  end
```

Comment:
Is it worth adding this to the view?

Reply:
Yes, I think so. Thanks for flagging this, I meant to come back to it. I don't really like this concatenation approach, in part because it will need a special case in the frontend URL obfuscation code (since it expects IDs to be numeric).

Is it a bad idea to just do some math to effectively prefix each ID with a fixed number? Like, if the record is a HUD Service it is prefixed with `1`, and if it's a Custom Service it's prefixed with a `2`. That would preserve uniqueness while still being numeric.

Reply:
Prefixing should work until we get up around the max value the integer field.

Reply:
I implemented the prefixing approach. I also updated the mutations to work when operating on the prefixed ID. Not too worried about that code through because the mutations will disappear once we switch to updating Services via form submission (the processor handles creation/updating).